### PR TITLE
[New Rule] Okta Brute Force or Password Spraying Attack

### PR DESCRIPTION
**Pull request by threat-punter**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/46

## Summary
Identifies a high number of failed Okta user authentication attempts from a single IP address, which could be indicative of a brute force or password spraying attack. An adversary may attempt a brute force or password spraying attack to obtain unauthorized access to user accounts.

See https://github.com/elastic/detection-rules/issues/46 for further information.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/okta/credential_access_okta_brute_force_or_password_spraying.toml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+updated_date = "2020/07/16"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies a high number of failed Okta user authentication attempts from a single IP address, which could be indicative
+of a brute force or password spraying attack. An adversary may attempt a brute force or password spraying attack to
+obtain unauthorized access to user accounts.
+"""
+index = ["filebeat-*"]
```
Comment (brokensound77) [inline rules/okta/credential_access_okta_brute_force_or_password_spraying.toml:14]:
```suggestion
false_positives = ["Automation with expired credentials and unbounded retries may lead to false positives."]
index = ["filebeat-*"]
```

File: rules/okta/credential_access_okta_brute_force_or_password_spraying.toml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+updated_date = "2020/07/16"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies a high number of failed Okta user authentication attempts from a single IP address, which could be indicative
+of a brute force or password spraying attack. An adversary may attempt a brute force or password spraying attack to
+obtain unauthorized access to user accounts.
+"""
+false_positives = ["Automation with expired credentials and unbounded retries may lead to false positives."]
```
Comment (threat-punter) [inline rules/okta/credential_access_okta_brute_force_or_password_spraying.toml:14]:
```suggestion
false_positives = ["Automated processes that attempt to authenticate using expired credentials and unbounded retries may lead to false positives."]
```

File: rules/okta/credential_access_okta_brute_force_or_password_spraying.toml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+description = """
+Identifies a high number of failed Okta user authentication attempts from a single IP address, which could be indicative
+of a brute force or password spraying attack. An adversary may attempt a brute force or password spraying attack to
+obtain unauthorized access to user accounts.
+"""
+false_positives = ["Automated processes that attempt to authenticate using expired credentials and unbounded retries may lead to false positives."]
+index = ["filebeat-*"]
+language = "kuery"
+license = "Elastic License"
+name = "Okta Brute Force or Password Spraying Attack"
```
Comment (peasead) [inline rules/okta/credential_access_okta_brute_force_or_password_spraying.toml:18]:
```suggestion
name = "Okta Brute Force or Password Spraying Attack"
note = "This requires enabling the Okta module for Filebeat."
```

File: rules/okta/credential_access_okta_brute_force_or_password_spraying.toml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+
+[rule.threat.tactic]
+id = "TA0006"
+name = "Credential Access"
+reference = "https://attack.mitre.org/tactics/TA0006/"
+
+[rule.threshold]
+field = "source.ip"
+value = 25
```
Comment (peasead) [inline rules/okta/credential_access_okta_brute_force_or_password_spraying.toml:49]: Just a comment, this seems like a lot of failed attempts. If this is a good measuring stick based on your experience, no need to change it - I've not investigated how many times a brute force/spray happens before success.

File: rules/okta/credential_access_okta_brute_force_or_password_spraying.toml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+Identifies a high number of failed Okta user authentication attempts from a single IP address, which could be indicative
+of a brute force or password spraying attack. An adversary may attempt a brute force or password spraying attack to
+obtain unauthorized access to user accounts.
+"""
+false_positives = ["Automated processes that attempt to authenticate using expired credentials and unbounded retries may lead to false positives."]
+index = ["filebeat-*"]
+language = "kuery"
+license = "Elastic License"
+name = "Okta Brute Force or Password Spraying Attack"
+note = "This requires enabling the Okta module for Filebeat."
```
Comment (threat-punter) [inline rules/okta/credential_access_okta_brute_force_or_password_spraying.toml:19]:
```suggestion
note = "The Okta Filebeat module must be enabled to use this rule."
```

File: rules/okta/credential_access_okta_brute_force_or_password_spraying.toml (hunk 4 above)
Comment (threat-punter) [reply rules/okta/credential_access_okta_brute_force_or_password_spraying.toml:50]: Thanks @peasead. The default lockout policy for Okta users is 10 failed authentication attempts. If an automated or brute force attack is happening, I expect the number of failed authentication attempts to be quite high.

I'm planning on developing another couple of rules to detect attempts to gain unauthorized access to Okta accounts. E.g. https://github.com/elastic/detection-rules/issues/73 and https://github.com/elastic/detection-rules/issues/74. We can always revisit this threshold value.

**Pull request merged**